### PR TITLE
Spike/redis

### PR DESCRIPTION
**Pull request by daiLlew**
**DO NOT MERGE**
PR just used to show spike code.

- Basic Cache interface for Setting & Getting Identities / auth tokens. 
- IdentityCache implementation is for `Redis`.
- Uses `MULTI` to update the TTL and Get the identity in a single transaction.

Rest of changes can be ignored and were just required to be able to test the impl.

### CODE REVIEW

File: api/ping.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package api
```
Comment (daiLlew) [inline api/ping.go:1]: Ignore this handler - I wrote it to be able to test that caching was working.

File: api/token.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ func (api *API) CreateTokenHandler(w http.ResponseWriter, r *http.Request) {
 func (api *API) createToken(ctx context.Context, tokenReq *NewTokenRequest) (*AuthToken, error) {
 	logD := log.Data{"email": tokenReq.Email}
 
-	err := api.IdentityService.VerifyPassword(ctx, tokenReq.Email, tokenReq.Password)
+	i, err := api.IdentityService.VerifyPassword(ctx, tokenReq.Email, tokenReq.Password)
```
Comment (daiLlew) [inline api/token.go:60]: Possibly need to consider updating this method to return the identity object as its needed when we cache the token

File: api/models.go [hunk 3]
```diff
@@ -81,5 +82,10 @@ func getNewTokenRequest(ctx context.Context, r io.ReadCloser) (*NewTokenRequest,
 //IdentityService is a service for creating, updating and deleting Identities.
 type IdentityService interface {
 	Create(ctx context.Context, i *identity.Model) (string, error)
-	VerifyPassword(ctx context.Context, email string, password string) error
+	VerifyPassword(ctx context.Context, email string, password string) (*identity.Model, error)
+}
+
+type Cache interface {
+	Set(token string, i identity.Model) error
+	Get(token string) (*identity.Model, error)
```
Comment (daiLlew) [inline api/models.go:90]: Basic functionality but also need to consider `Delete` and `DeleteAll` scenarios as well.

File: cache/cache.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+type IdentityCache struct {
+	pool redis.Pool
+	ttl  int
+}
+
+func (c *IdentityCache) Set(token string, i identity.Model) error {
+	conn := c.pool.Get()
+	defer conn.Close()
+
```
Comment (daiLlew) [inline cache/cache.go:42]: Redis does not support `GET`and `UPDATE TTL` in a single call. To achieve this we need to use `MULTI` todo both actions in a transaction.

**Comment by janderson2** [comment]
Repo is archived. Closing

**Pull request closed without merging**